### PR TITLE
ci: test bundle-upgrade scenario

### PR DESCRIPTION
**Pull request by sthaha**
Previously, in CI, the operator bundle was installed on a clean cluster instead of the scenario where the bundle is upgraded. This has resulted in bundle getting rejected in Community Operator Catalog when the upgrade test failed.

This commit modifies the e2e test to run the upgrade scenario instead of installing on a fresh cluster.

### CODE REVIEW

File: tests/run-e2e.sh [hunk 1]
```diff
@@ -89,19 +92,36 @@ gather_olm() {
...
+		line 50 heavy
 		gather_olm || true
+		fail "Running Released Bundle failed"
+		return $ret
 	}
 
-	return $ret
+	info "Running Upgrade to new bundle"
+	run ./tmp/bin/operator-sdk run bundle-upgrade "$BUNDLE_IMG" \
+		--namespace "$OPERATORS_NS" --use-http || {
```
Comment (vprashar2929) [inline tests/run-e2e.sh:116]: why use `--use-http` ?

Comment (sthaha) [reply tests/run-e2e.sh:116]: because localhost:5000 is using http

Comment (vprashar2929) [reply tests/run-e2e.sh:116]: oh got it

Comment (sthaha) [reply tests/run-e2e.sh:116]: https://github.com/operator-framework/operator-sdk/blob/master/internal/olm/operator/registry/index_image.go#L131

**Review by vprashar2929 — APPROVED** [review]

**Pull request merged**